### PR TITLE
Optimization hardened

### PR DESCRIPTION
**Pull request by Andarwinux**
Built and tested.

When using the clang toolchain,
enable Intel CET and Microsoft ControlFlowGuard and LLVM Polly optimization for x86_64.
enable Microsoft ControlFlowGuard and LLVM Polly optimization for aarch64.

### CODE REVIEW

**Comment by Andarwinux** [comment]
There seems to be a problem, currently no polly flags are passed to lld, so polly doesn't actually take effect if lto is enabled.
Also, llvm-ld.in doesn't actually work.
This will be fixed later.

**Comment by Andarwinux** [comment]
Although only mpv is enable thinlto, there are future plans to implement the optional full-packages LTO option. Since lld 17 does not currently support `-ffat-lto-objects` in the COFF backend, this feature will have to wait until llvm18 is released.

File: toolchain/llvm/llvm-wrapper.cmake [hunk 1]
```diff
@@ -23,6 +23,7 @@ ExternalProject_Add(llvm-wrapper
     COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_INSTALL_PREFIX}/bin/llvm-addr2line ${CMAKE_INSTALL_PREFIX}/bin/${TARGET_ARCH}-addr2line
     COMMAND ${CMAKE_COMMAND} -E create_symlink ${PKGCONFIG} ${CMAKE_INSTALL_PREFIX}/bin/${TARGET_ARCH}-pkg-config
     COMMAND ${CMAKE_COMMAND} -E create_symlink ${PKGCONFIG} ${CMAKE_INSTALL_PREFIX}/bin/${TARGET_ARCH}-pkgconf
+    COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_INSTALL_PREFIX}/bin/${TARGET_ARCH}-ld ${CMAKE_INSTALL_PREFIX}/bin/ld.${TARGET_ARCH}-lld
```
Comment (shinchiro) [inline toolchain/llvm/llvm-wrapper.cmake:26]: When `ld.${TARGET_ARCH}-lld` will be used?

Comment (shinchiro) [inline toolchain/llvm/llvm-wrapper.cmake]: Moved the ld symlink before pkg-config for consistency

**Comment by shinchiro** [comment]
LGTM

**Comment by shinchiro** [comment]
ffmpeg doesnt actually use polly flags in config.log

**Comment by Andarwinux** [comment]
Not using Polly during the configure phase is the expected behavior. During this phase, only compile tests are conducted, it's possible that certain test items have added flags like `-O0` or similar.

**Comment by shinchiro** [comment]
So how to check the flags is really used by compiler?
Also rebase this PR too

**Comment by Andarwinux** [comment]
> So how to check the flags is really used by compiler? Also rebase this PR too

`-fsave-optimization-record`

**Pull request merged**